### DISCUSSION TITLE
cell base chaste memory limits

### DISCUSSION THREAD

**Discussion by Elphikaus** [Chaste/Chaste#176] [Q&A] [answered]
## TL/DR;

My main question: how can I adjust how much memory I allow Chaste to use?

I would like to increase this amount as I have 32GB of RAM and I think Chaste kills simulations well before it uses all of what's available.

## Longer version
I am running some off-lattice node-based cell models.

My cell populations divide and run for some $N$ cell cycles and I generally end up with a few hundred cells.

This usually means that my simulation stop time is approximately 7 (I think the units are hours), where each cell cycle is uniform around 1.

When I try to run my simulations past 7 or so, I get errors saying my simulation threw an exception.

I am not sure how to gauge what is happening, but I wanted to start by exploring the memory case first.

Also, I have 8 cores that I can multi-thread to 16, is there any way I can parallelise multiple runs of a model?

**Comment by jmpf** [comment, marked as answer]
Apologies that I've only just read this question.

There shouldn't be reason for Chaste to crash because it's out of memory any time before you run out of physical memory/swap.  Chaste memory management is all via traditional heap-based allocation.  Is it the case that you're in a docker container?  If so, we should add docker to title (and/or investigate how to increase the default limits in docker).

As for the multi-threaded question: there are some overlapping spheres simulations that can be run with parallelisation.  This is the only cell-based paradigm that has been parallelised.  Some examples of the capability were given in https://doi.org/10.1016/j.cpc.2015.03.005  
If you look in the cell_based/test/simulation folder for test suites with #include "PetscSetupAndFinalize.hpp" rather than #include "FakePetscSetup.hpp" then you might find an example similar to what you want to do.

**Comment by jmpf** [comment]
> Apologies that I've only just read this question.

I think that I should thank @mirams who converted it from an issue.
